healthcheck/client_test.go

```diff
@@ -0,0 +1,59 @@
+package healthcheck
+
+import (
+	"errors"
+	"net/http"
+	"testing"
+
+	"github.com/ONSdigital/go-ns/healthcheck/mock_healthcheck"
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+func TestUnitHealthcheckClient(t *testing.T) {
+
+	service := "myService"
+
+	Convey("Healthcheck returns nil when no error", t, func() {
+		mock := &mock_healthcheck.HttpClientMock{
+			GetFunc: func(url string) (*http.Response, error) {
+				return &http.Response{StatusCode:http.StatusOK}, nil
+			},
+		}
+
+		client := healthcheckClient{Client:mock, Url:"http://foo/bar", Service: service}
+
+		s, e := client.Healthcheck()
+		So(s, ShouldEqual, service)
+		So(e, ShouldBeNil)
+	})
+
+	Convey("Healthcheck returns error when status != 200", t, func() {
+		mock := &mock_healthcheck.HttpClientMock{
+			GetFunc: func(url string) (*http.Response, error) {
+				return &http.Response{StatusCode:http.StatusInternalServerError}, nil
+			},
+		}
+
+		client := healthcheckClient{Client:mock, Url:"http://foo/bar", Service: service}
+
+		s, e := client.Healthcheck()
+		So(s, ShouldEqual, service)
+		So(e, ShouldNotBeNil)
+	})
+
+	Convey("Healthcheck returns error when Get returns error", t, func() {
+		err := errors.New("This is an error")
+		mock := &mock_healthcheck.HttpClientMock{
+			GetFunc: func(url string) (*http.Response, error) {
+				return &http.Response{StatusCode:http.StatusInternalServerError}, err
+			},
+		}
+
+		client := healthcheckClient{Client:mock, Url:"http://foo/bar", Service: service}
+
+		s, e := client.Healthcheck()
+		So(s, ShouldEqual, service)
```

Comment:
The generated mocks also maintain a list of calls to each function on the mock. Its quite common to assert the number of calls to a mock function using: `So(len(mock.GetCalls()), ShouldEqual, 1)`